Sources/TokamakCore/Modifiers/AspectRatioLayout.swift

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020-2021 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+@frozen public enum ContentMode: Hashable, CaseIterable {
+  case fit
+  case fill
+}
+
+public struct _AspectRatioLayout: ViewModifier {
+  public let aspectRatio: CGFloat?
+  public let contentMode: ContentMode
+
+  @inlinable
+  public init(aspectRatio: CGFloat?, contentMode: ContentMode) {
+    (self.aspectRatio, self.contentMode) = (aspectRatio, contentMode)
```

Comment:
nit: from coding style perspective I'd prefer it to be on separate lines, just for consistency's sake with the rest of the codebase.